hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/HandleContext.java

```diff
@@ -71,4 +74,28 @@ public interface HandleContext {
      */
     @NonNull
     <C> C createReadableStore(@NonNull Class<C> storeInterface);
+
+    /**
+     * Gets the {@link SignatureVerification} for the given key. If this key was unknown during signature verification,
+     * either because it was not provided as a required key or because the signature map did not have a "full prefix"
+     * for this key, then {@code null} is returned.
+     *
+     * @param key the key to get the verification for
+     * @return the verification for the given key, or {@code null} if no known key or signature at pre-handle
+     */
+    @Nullable
+    SignatureVerification verificationFor(@NonNull final Key key);
```

Comment:
Note (probably for a future PR):
If we keep this single verification approach, the method the signature verification code uses to walk the tree should be exposed in a utility, so other code (e.g. ScheduleSign) that needs to take a set of keys and decompose that into a flat list of many possibly optional cryptographic keys can do so without duplicating the same complex code.
For example:
Schedule Sign dispatches a pre-handle for a Token Transfer internal transaction that requires 8 signatures, two are keylist, three are MofN threhsolds, and three are single keys.  ScheduleSign then needs to get all of the individual keys (all the way down to the cryptographic keys under those "keys") and add them as optional keys on the context because any subset of individual cryptographic keys might have signed the current transaction.

Rather than recreate the code to walk a potentially extensive grove of trees of keys and subkeys in Schedule Service, it would be good to have a utility method that can be called to get all of the individual keys that is then shared from the verification code.

Reply:
I agree, we need to find a way to share that logic. Good thing for follow up PR.